components/dashboard/context/DashboardContext.tsx

```
@@ -0,0 +1,147 @@
+
+interface DashboardContextType {
+  // User dashboard preferences
+  preferences: DashboardPreferences;
+  updatePreferences: (newPrefs: Partial<DashboardPreferences>) => void;
+  
+  // Dashboard state
+  isSidebarOpen: boolean;
+  toggleSidebar: () => void;
+  
+  // Dashboard data
+  recentActivity: any[]; // Replace with proper types when implementing actual activity tracking
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)